src/som/interpreter/nodes/literals/ObjectLiteralNode.java

```diff
@@ -30,7 +32,20 @@ public ObjectLiteralNode(final MixinDefinition mixinDef, final ExpressionNode ne
 
   @Override
   public Object executeGeneric(final VirtualFrame frame) {
-    SObjectWithClass outer = (SObjectWithClass) SArguments.rcvr(frame);
+    /* Obtain the outer object from the receiver, which is either the receiver
+     * itself (in cases where the receiver is a class) or the `outerSelf`
+     * (in cases where the receiver is a block).
+     */
+    SAbstractObject rcvr = (SAbstractObject) SArguments.rcvr(frame);
+    SObjectWithClass outer;
+    if (rcvr instanceof SObjectWithClass) {
+      outer = (SObjectWithClass) rcvr;
+    } else if (rcvr instanceof SBlock) {
+      outer = (SObjectWithClass) ((SBlock) rcvr).getOuterSelf();
+    } else {
```

Comment:
why don't you just use a node here that does the right thing?
There should a node to read outer, no need to duplicate that logic here, or is it somehow special?